cli/cmd/inject.go

```diff
@@ -179,6 +179,7 @@ func injectPodTemplateSpec(t *v1.PodTemplateSpec, controlPlaneDNSNameOverride, v
 				Name:  "CONDUIT_PROXY_DESTINATIONS_AUTOCOMPLETE_FQDN",
 				Value: "Kubernetes",
 			},
+			v1.EnvVar{Name: "CONDUIT_PROXY_DEPLOYMENT", Value: owner},
```

Comment:
I thought we are trying to keep the proxy code as oblivious to Kubernetes as possible. If so, shouldn't we pass in a more full string here, instead of just the deployment name, so that we don't have to write Kubernetes-specific code in the proxy to join the deployment name with the namespace and maybe other string parts to form the string value that should be reported in metrics?

Reply:
Perhaps. Although the proxy already needs to know its namespace to do fqdn auto-completion...

Reply:
> Perhaps. Although the proxy already needs to know its namespace to do fqdn auto-completion...

Yes, that's true. But I think we should try to isolate the Kubernetes-specific knowledge to service discovery, if we can, and hopefully even remove it from there.

Reply:
I agree.  However, the namespace is not necessarily known at injection time.  This can be worked around with the downward API and expanding one env var into another but this is more complex and harder to test.  I'd prefer to leave this as is for now and revisit when we change the fqdn code.

Reply:
> I agree. However, the namespace is not necessarily known at injection time. This can be worked around with the downward API and expanding one env var into another but this is more complex and harder to test. I'd prefer to leave this as is for now and revisit when we change the fqdn code.

You mean because a default namespace might be used? i.e. `kuebectl apply -f -n some-other-default` would cause the default namespace to be `some-other-default` instead of `default`? That's a good point I hadn't considered. We should document that in a comment in the code so that we don't lose that detail.